libs/community/langchain_community/document_loaders/sql_database.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any, Callable, Dict, Iterator, List, Optional, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+from langchain_community.utilities.sql_database import SQLDatabase
+
+
+class SQLDatabaseLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+
+    For talking to the database, the document loader uses the `SQLDatabase`
+    utility from the LangChain integration toolkit.
+
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        db: SQLDatabase,
+        parameters: Optional[Dict[str, Any]] = None,
```

Comment:
Could we add `*` after db, to force named argument invocation for all optionals (it helps  to minimize breaking changes during future refactors of the API)